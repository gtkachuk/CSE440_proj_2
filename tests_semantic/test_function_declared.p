program testFunctionDeclared;

class testFunctionDeclared

BEGIN
   
   VAR
      lightSwitch   : boolean;
      compilerWorks : boolean;
      value1	    : boolean;
      value2	    : boolean;

FUNCTION setCompilerWorks;
BEGIN
   lightSwitch := value1
END ;

FUNCTION setCompilerWorks;
BEGIN
   compilerWorks := value2
END ;

FUNCTION testFunctionDeclared;
BEGIN
   lightSwitch := compilerWorks
END

END
.
