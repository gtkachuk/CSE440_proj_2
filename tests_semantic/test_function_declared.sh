#!/bin/sh

. ../tests_lib.inc

test_diff_sem test_function_declared.p
