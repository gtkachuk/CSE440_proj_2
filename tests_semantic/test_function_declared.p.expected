ERROR:21:Function "setCompilerWorks" already declared at line 16
Errors detected. Exiting.
